app/models/account.rb

```diff
@@ -1,9 +1,19 @@
 # Customer organization account
 class Account < ActiveRecord::Base
+  # @param [String] piece the tenant piece of the canoncial name
+  # @return [String] full canonical name
+  # @see Settings.multitenancy.default_host
+  def self.default_cname(piece)
+    return unless piece
+    default_host = Settings.multitenancy.default_host
+    format(default_host, tenant: piece.parameterize)
+  end
+
   attr_readonly :tenant
-  validates :name, presence: true
+  # name is unused after create, only used by sign_up/new forms
+  validates :name, presence: true, unless: 'cname.present? && cname != default_cname("")'
```

Comment:
However, changing the default does introduce some weirdness since the previous instance-level `default_cname` param takes a different default.  I find the explicit args help make it clear what the expected result is since the logic is the same.  We're talking about 2 calls, so I don't think this warrants much attention.